### ISSUE TITLE
IndexError when building documentation 

### ISSUE THREAD

**Issue by adrianbroher** [michaeljones/breathe#286]
When testing out breathe and sphinx to prepare the merge of C++ and python API documentation I stumbled over an IndexError when building the doc.

Doxygen version: 1.8.11
Sphinx version: 1.4.8
Breathe version: 4.2.0

I'm sorry but for now I can't give any more information as I just implemented the quick start guides for both sphinx and breathe so any hint on how to get more information out of the software would be much appreciated.

Stack trace:
```
$ cat /tmp/sphinx-err-dW8QU2.log 
# Sphinx version: 1.4.8
# Python version: 2.7.12 (CPython)
# Docutils version: 0.12 repository
# Jinja2 version: 2.8
# Last messages:
#   Running Sphinx v1.4.8
#   loading pickled environment...
#   done
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 1 source files that are out of date
#   updating environment:
#   0 added, 1 changed, 0 removed
#   reading sources... [100%] index
# Loaded extensions:
#   breathe (4.2.0) from /usr/lib/python2.7/site-packages/breathe/__init__.pyc
#   alabaster (0.7.9) from /usr/lib/python2.7/site-packages/alabaster/__init__.pyc
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/sphinx/cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "/usr/lib/python2.7/site-packages/sphinx/application.py", line 297, in build
    self.builder.build_update()
  File "/usr/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 251, in build_update
    'out of date' % len(to_build))
  File "/usr/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 265, in build
    self.doctreedir, self.app))
  File "/usr/lib/python2.7/site-packages/sphinx/environment.py", line 569, in update
    self._read_serial(docnames, app)
  File "/usr/lib/python2.7/site-packages/sphinx/environment.py", line 589, in _read_serial
    self.read_doc(docname, app)
  File "/usr/lib/python2.7/site-packages/sphinx/environment.py", line 742, in read_doc
    pub.publish()
  File "/usr/lib/python2.7/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/usr/lib/python2.7/site-packages/sphinx/io.py", line 49, in read
    self.parse()
  File "/usr/lib/python2.7/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/__init__.py", line 172, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 170, in run
    input_source=document['source'])
  File "/usr/lib/python2.7/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2726, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 395, in new_subsection
    node=section_node, match_titles=True)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 282, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python2.7/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/lib/python2.7/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2299, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2311, in explicit_construct
    return method(self, expmatch)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2054, in directive
    directive_class, match, type_name, option_presets)
  File "/usr/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2103, in run_directive
    result = directive_instance.run()
  File "/usr/lib/python2.7/site-packages/breathe/directive/index.py", line 91, in run
    return self.handle_contents(project_info)
  File "/usr/lib/python2.7/site-packages/breathe/directive/index.py", line 60, in handle_contents
    node_list = object_renderer.render()
  File "/usr/lib/python2.7/site-packages/breathe/renderer/index.py", line 14, in render
    nodelist.extend(compound_renderer.render())
  File "/usr/lib/python2.7/site-packages/breathe/renderer/index.py", line 63, in render
    rendered_data = data_renderer.render()
  File "/usr/lib/python2.7/site-packages/breathe/renderer/compound.py", line 39, in render
    return compound_renderer.render()
  File "/usr/lib/python2.7/site-packages/breathe/renderer/compound.py", line 123, in render
    child_nodes = renderer.render()
  File "/usr/lib/python2.7/site-packages/breathe/renderer/compound.py", line 160, in render
    node_list.extend(renderIterable(self, self.data_object.memberdef))
  File "/usr/lib/python2.7/site-packages/breathe/renderer/compound.py", line 21, in renderIterable
    output.extend(child_renderer.render())
  File "/usr/lib/python2.7/site-packages/breathe/renderer/compound.py", line 239, in render
    self.update_signature(signode)
  File "/usr/lib/python2.7/site-packages/breathe/renderer/compound.py", line 410, in update_signature
    signode.children[1][0] = self.node_factory.strong(text="[anonymous]")
IndexError: list index out of range
```

**Comment by AWhetter** [comment]
I'm getting a similar exception, although I think the location has changed.

```
breathe/renderer/sphinxrenderer.py", line 935, in update_signature
    signature.children[1][0] = self.node_factory.strong(text="[anonymous]")
IndexError: list index out of range
```

Seems to happen when parsing the documentation for an anonymous enum on a class for me.

```
namespace MyNamespace {
class MyClass {
  public:
    enum {color = 1};
}
}
```

**Comment by aldanor** [comment]
This is same as #308. Any ideas?

**Comment by melvinvermeeren** [comment]
Although usually the oldest usually is kept open, is this case the actual fix was done by @michaeljones through issue #313. Closing this in favour of the already fixed #313.
